partner/app/tests/test_app.py

```diff
@@ -92,3 +92,51 @@ def test_read_tilesp_validate_sub2(client, sub2):
     assert "status" in response_content
     assert "count" in response_content
     assert "response" in response_content
+
+
+@pytest.mark.parametrize(
+    "country_code",
+    [
+        "invalid-param",
+        "us",
+        "11",
+        "1U",
+        "USAC",
+        "🛒📈🤖",
+    ],
+    ids=[
+        "hyphen_in_value",
+        "all_lowercase",
+        "numeric",
+        "alpha_numeric",
+        "exceeds_max_characters",
+        "emoji",
+    ],
+)
+def test_read_tilesp_validate_country_code(client, country_code):
+    """Test that only two uppercase characters are
+    accepted as values for the country code query parameter.
+    See https://github.com/mozilla-services/contile-integration-tests/issues/39
+    """
+    response = client.get(
+        "/tilesp",
+        params={
+            "partner": "demofeed",
+            "sub1": "123456789",
+            "sub2": "sub2",
+            "country-code": country_code,
+            "region-code": "NY",
+            "form-factor": "desktop",
+            "os-family": "macos",
+            "v": "1.0",
+            "results": "2",
+        },
+    )
+
+    assert response.status_code == 400
```

Comment:
With this pull request the only test coverage for accepted country-codes is `US` in `test_read_tilesp` and `test_read_tilesp_validate_sub2`. I suggest we add a new parameterized test to verify a set of accepted country-codes (for instance the regions which are currently included in the cloud operations infrastructure configuration).